kratos/utilities/geometry_utilities.h

```diff
@@ -821,13 +676,17 @@ class GeometryUtils
 
     /**
      * @brief Calculate the deformation gradient.
-     * 
-     * See, e.g., P. Wriggers, Nonlinear Finite Element Methods, Springer, 2008.
+     * @details See, e.g., P. Wriggers, Nonlinear Finite Element Methods, Springer, 2008.
      * @param rJ element Jacobian.
      * @param rInvJ0 inverse of the element Jacobian of the initial configuration.
      * @param rF deformation gradient.
      */
-    static void DeformationGradient(Matrix const& rJ, Matrix const& rInvJ0, Matrix& rF)
+    template<class TMatrix1, class TMatrix2, class TMatrix3>
+    static void DeformationGradient(
+        TMatrix1 const& rJ,
+        TMatrix2 const& rInvJ0,
+        TMatrix3& rF
+        )
     {
         if (rF.size1() != rJ.size1() || rF.size2() != rInvJ0.size2())
             rF.resize(rJ.size1(), rInvJ0.size2(), false);
```

Comment:
this will not compile with AMatrix, same issue that was already discussed in the previous days

Reply:
True

Reply:
I "fixed" on https://github.com/KratosMultiphysics/Kratos/pull/4144/commits/6d86cb2446f895d464fd300ca5a4c8b467389219 sincerely is difficult to tackle that